### ISSUE TITLE
Error on share

### ISSUE THREAD

**Issue by mwarkentin** [roidrage/s3itch#4]
Hey, I think I've configured everything the way it's supposed to be, but I'm getting this error:

```
**Webpost error**

Posted file is not appearing where it should.
Please check your "Base URL" setting
```

The weird part is that the files are showing up in the bucket..

**Comment by roidrage** [comment]
Did you set up S3 bucket the same on for the app and locally for Skitch? This sounds like there may be an inconsisency issue here, which is not extremely unlikely, but in general S3 promises that you can read your own writes. Did you by any chance set up a reduced redundancy bucket for this?

**Comment by mwarkentin** [comment]
Looks like I fixed this.. I decided to try it without the user / pass settings, and it seems to work correctly now. Here are some of the details I was going to send, if that would help you figure out what was going on..

I created a bucket called `skitch.michaelwarkentin.com`, I believe it's a normal bucket (not reduced redundancy), but I'm not sure how to make sure.

Here are my Skitch settings:

![Skitch Settings](http://f.cl.ly/items/1X0E2N0a0U1w3n231s0o/Preferences.png)

**Comment by roidrage** [comment]
Weird. The guy who implemented HTTP auth didn't seem to have any problems with this.
The only oddity that I can see now is the S3 bucket URL, shouldn't have to be this long, and I'd recommend adding a CNAME record for the bucket so that it's just skitch.michaelwarkentin.com.

**Comment by lstoll** [comment]
If credentials aren't specified, any auth headers will just be ignored so this shouldn't be an issue. As well, that error comes up after the file is uploaded, and it tries to read it from S3 - which this code doesn't touch.

I have seen this error personally - I think it's a filename encoding issue. Some characters aren't being correctly escaped at some point, so the target filename doesn't match the URL skitch is expecting. I've been meaning to look at this, but it hasn't bothered me enough yet. I just edit the filename.

**Comment by roidrage** [comment]
Yeah, I noticed this issue too, comes up with weird website titles for examples. Need to try and reproduce and watch what part of the whole setup specifically fails.

**Comment by matiaskorhonen** [comment]
My issue was caused by an incorrect bucket name.

<del>I also have this problem and I get this in the Skitch logs when trying to share (any) file:</del>


```
**** HTTP URL CHECK VIA CURL ****

* About to connect() to mys3bucketname.s3-website-eu-west-1.amazonaws.com port 443 (#0)
*   Trying 178.236.4.60... * Connection refused
* couldn't connect to host
* Closing connection #0
* Couldn't connect to server
```

<del>I get the same result regardless of whether I try using HTTP or HTTPS for the base URL.</del>

**Comment by matiaskorhonen** [comment]
Issue #9 is probably related to this.

**Comment by matiaskorhonen** [comment]
I've noticed that hash or pound symbols (`#`) don't get escaped by Skitch and thus cause an error on upload.

**Comment by matiaskorhonen** [comment]
PR #10 _might_ fix some of these issues because at least the case where the title has a hash blows up on the Content-Type identification at the moment.

**Comment by roidrage** [comment]
Totally missed #10, sorry. Going to look into it!

**Comment by matiaskorhonen** [comment]
I just added a fix for hashes in the title, too.
